Sources/RTMP/RTMPConnection.swift

```diff
@@ -456,6 +458,19 @@ open class RTMPConnection: EventDispatcher {
         previousTotalBytesIn = totalBytesIn
         previousTotalBytesOut = totalBytesOut
         previousQueueBytesOut.append(socket.queueBytesOut.value)
+        
+        if currentBytesOutPerSecond == 0 {
+            zeroBytesOutCounter += 1
+            // If currentBytesOutPerSecond has been 0 for the last 5 seconds
+            if zeroBytesOutCounter >= 5 {
+                print("STALE connection, force close")
+                self.close(isDisconnected: true) // call close function
+                zeroBytesOutCounter = 0 // reset counter
+            }
+        } else {
+            zeroBytesOutCounter = 0 // reset counter
+        }
```

Comment:
I believe it is abnormal for there to be no output for 5 seconds during publishing. On the other hand,  Playback feature it is a possible scenario. If I accept it as it is, it will be considered a bug.

Reply:
It seems like a good idea to implement the PING/PONG mechanism for RTMP.

Reply:
Yes  maybe ping pong is a better idea, I couldn't tell if it's standard for RTMP.
For me the stale connections happen with this output:

2023-06-22 16:10:31.674537+0100 appnew[1163:121214] [tcp] tcp_input [C15:1] flags=[R] seq=1983586932, ack=0, win=0 state=CLOSE_WAIT rcv_nxt=1983586933, snd_una=1331617881


CLOSE_WAIT means that the local end of the connection has received a FIN from the other end, but the OS is waiting for the program at the local end to actually close its connection. The problem is your program running on the local machine is not closing the socket.

After CLOSE_WAIT notice from NWConnection provided by OS, RTMP Connection is still active and sends publishSufficientBWOccured,  but currentBytesOutPerSecond are 0  and it can last for long time  1+ minutes.

I don't know where CLOSE_WAIT is coming from.. maybe server issue. Tried with different servers,  RTMP, RTMPS, with balancer, and without, but still getting CLOSE_WAIT when reconnecting quickly.